### ISSUE TITLE
Can't access values of Measurement as property when `dask>=2024.11.0`

### ISSUE THREAD

**Issue by emmaai** [opendatacube/datacube-core#1656]
### Expected behaviour
Measurement values can be access as property e.g., by `measurement.dtype`.
The symptom is that somehow (and somewhere) `Measurement` is casted back to parent class `dict` instead of `datacube.model.Measurement`. Logging parameter `type(measurement)`  in `fuse_lazy` shows the different classes information between `dask==2024.10.0` and `dask==2024.11.0`

### Actual behaviour
Emit error when compute 
```
Exception: 'AttributeError("\'dict\' object has no attribute \'nodata\'")'
Traceback: '  File "/env/lib/python3.10/site-packages/datacube/api/core.py", line 946, in fuse_lazy\n    print(f\'measurement nodata {measurement.nodata}\')\n'
```

### Steps to reproduce the behaviour
Loading multiple scenes  with `groupby="solar_day"`

### Environment information

* Which ``datacube --version`` are you using? 
1.8.19
* What datacube deployment/enviornment are you running against? 
sandbox

> **Note:** Stale issues will be automatically closed after a period of six months with no activity. 
> To ensure critical issues are not closed, tag them with the Github `pinned` tag.
> If you are a community member and not a maintainer please escalate this issue to maintainers via
> [GIS StackExchange](https://gis.stackexchange.com/questions/tagged/open-data-cube) or [Discord](https://discord.com/invite/4hhBQVas5U).

**Comment by emmaai** [comment]
tag the issue for tracking https://github.com/dask/distributed/issues/8932

**Comment by omad** [comment]
Thanks Emma. Love the reproducible example you posted in the dask issue!

**Comment by omad** [comment]
And then I promptly forgot, and spent far too much time today wondering why `datacube-core` storage test cases were failing for me! 😢

**Comment by SpacemanPaul** [comment]
It's particularly bad with numpy2.   

2024.11.1 works with numpy1 - but not numpy2, and 2024.11.2 breaks both again.

**Comment by SpacemanPaul** [comment]
There are a bunch of other more or less subtle upstream library issues with numpy2.

I just got develop-1.9 CI running with numpy 2 in #1661 - required pinning the MOST recent versions of shapely, netCDF4, and cftime; installing netcdf4 with `--no-binary`; and pinning fairly specific recent versions of dask and pandas. 

Dask seems to be the only one affecting numpy 1 though.

**Comment by omad** [comment]
Re-opening for investigation. 

We can't simply wait for dask/distributed to fix this. Pinning to an increasingly old version is a work around, not a solution.

For example, having an up to date nvidia/cuda python toolchain requires specific versions of dask and distributed.

**Comment by emmaai** [comment]
I'm looking into this atm on the dask cuz I need a newer version of dask now. did you plan to fix it? then I can wait for you :)

**Comment by SpacemanPaul** [comment]
Was only intended as a stop-gap - I was expecting the competing regressions in dask/distributed to settle down reasonably soon, at which point we can go back to a minimum version pin - hopefully this has already happened on the dask side, but I haven't looked into it since late last year.

**Comment by omad** [comment]
@emmai I only just realised it was a problem, haven't gotten any further yet. Waiting on me is rarely a good plan, but if you haven't solved it I'll have a look. Where are you up to.

@SpacemanPaul Thanks Paul. I'd expected the same, but I can't see any activity on the dask/distributed side.

**Comment by omad** [comment]
Testing some more, it appears to be fixed as of dask/distributed 2024.12.1!

**Comment by emmaai** [comment]
somehow compatibility pinned me to `2024.11.1`. I was trying to see if there was any easy fix on the bug mentioned in the reply. good news if it was fixed in higher version!

**Comment by omad** [comment]
> somehow compatibility pinned me to `2024.11.1`. 

Strange!

I'm back on it today. Two test examples indicate that it's fixed in 2024.12.1+. But, datacube-core tests are still failing. There might be something else wrong too.

I'll keep you posted.

**Comment by SpacemanPaul** [comment]
Yeah, I seem to remember that there were two separate Dask issues at play - and they both went back and forth from broken to fixed to broken again in different versions.

**Comment by omad** [comment]
This is fixed in the 1.9.1 release.
